### ISSUE TITLE
Attachment added when start date is on/after a specific date (not supported calendar message.ics)

### ISSUE THREAD

**Issue by xrisdoc** [ical-org/ical.net#106]
I have managed to get a Calendar invite setup and working as expected. However, when I reach a specific date for the Start date and after for an event, I get an email with the following file attached:  "not supported calendar message.ics"

I am sending this using a multipart/alternative email. Adding a text/calendar alternative view with the relevant ICS content.

It seems to work as expected up until a specific start date and then after that start date, this is when it seems to get the email attachment included.

The code for this is as follows:

```
using (Calendar calendar = new Calendar())
{
    // DateTime eventDate = new DateTime(2016, 10, 30).ToUniversalTime(); // This seems to work fine
    DateTime eventDate = new DateTime(2016, 10, 31).ToUniversalTime(); // Using this date (and after) results in Attachment - "not supported calendar message.ics"
    calendar.Method = "REQUEST";

    // Create the event
    Event eventTraineeReview = new Event()
    {
        Uid = Guid.NewGuid().ToString("N"),
        Summary = "Test Event Invite",
        Description = "Test Event Description",
        Location = "Test Location",
        Start = new CalDateTime(eventDate),
        End = new CalDateTime(eventDate.AddHours(1)),
        Sequence = 0,
        Created = new CalDateTime(DateTime.Now),
        LastModified = new CalDateTime(DateTime.Now)
    };

    // Set the organizer of the event
    eventTraineeReview.Organizer = new Organizer()
    {
        CommonName = "Test Organizer",
        Value = new Uri("MAILTO:system@test.com")
    };

    // Create an alarm for the event
    Alarm alarmTraineeReview = new Alarm()
    {
        Action = AlarmAction.Display,
        Description = "REMINDER",
        Trigger = new Trigger()
        {
            Duration = TimeSpan.FromDays(-1),
            Related = TriggerRelation.Start
        }
    };

    // Add an alarm (reminder) for the event
    eventTraineeReview.Alarms.Add(alarmTraineeReview);

    // Add the event to the calendar
    calendar.Events.Add(eventTraineeReview);

    CalendarSerializer serializer = new CalendarSerializer();
    string icsContent = serializer.SerializeToString(calendar);

    // Save the calendar content to a file.
    string calendarSavePath = System.IO.Path.Combine("C:\\Temp\\", string.Format("{0}-IcsEmail.txt", DateTime.Now.ToString("yyyyMMddHHmmssfff")));
    System.IO.File.AppendAllText(calendarSavePath, icsContent);

    return serializer.SerializeToString(calendar);
}
```

I have also supplied two text files showing the resulting ICS content for each event date. It seems that the the formatting of the value for DTSTART and DTEND dates differ between the two.

[StartDate-31-10-2016-Ics.txt](https://github.com/rianjs/ical.net/files/434626/StartDate-31-10-2016-Ics.txt)
[StartDate-30-10-2016-Ics.txt](https://github.com/rianjs/ical.net/files/434625/StartDate-30-10-2016-Ics.txt)

This is happening in Outlook 2013.

**Comment by rianjs** [comment]
"Not supported calendar message" is an Outlook error. I used the following code, saved the output string to a file, and then used Outlook 2013 to import it. It looks like there's some weirdness going on with one of the serialization steps:
- Broken, from your Oct 31 attachment: `DTSTART;VALUE=DATE:20161031`
- Working (output from what I did below): `DTSTART:20161029T230000Z`

The idea you're trying to convey, I think, is "this DateTime I created is UTC". You do that by specifying the `DateTimeKind` value. `ToUniversalTime()` _does a time conversion from your local time zone._

Here's an example. On Oct 31, my local time zone will be UTC -4:

``` csharp
Console.WriteLine(new DateTime(2016, 10, 31).ToUniversalTime());
//Displays: 10/31/2016 4:00:00 AM -- not what you want

Console.WriteLine(new DateTime(2016, 10, 31, 0, 0, 0, DateTimeKind.Utc));
//Displays: 10/31/2016 12:00:00 AM -- this is what you actually want
```

I would replace that line with `var eventDate = new DateTime(2016, 10, 31, 0, 0, 0, DateTimeKind.Utc);`

This code imports ics files into Outlook properly:

``` csharp
var calendar = new Calendar();

// Using this date (and after) results in Attachment - "not supported calendar message.ics"
var eventDate = new DateTime(2016, 10, 31, 0, 0, 0, DateTimeKind.Utc);
calendar.Method = "REQUEST";

// Create the event
var eventTraineeReview = new Event
{
    Uid = Guid.NewGuid().ToString("N"),
    Summary = "Test Event Invite",
    Description = "Test Event Description",
    Location = "Test Location",
    Start = new CalDateTime(eventDate),
    End = new CalDateTime(eventDate.AddHours(1)),
    Sequence = 0,
    Created = new CalDateTime(DateTime.Now),
    LastModified = new CalDateTime(DateTime.Now),
    Organizer = new Organizer
    {
        CommonName = "Test Organizer",
        Value = new Uri("MAILTO:system@test.com")
    }
};

//Snip

var serializer = new CalendarSerializer(new SerializationContext());
var icsContent = serializer.SerializeToString(calendar);
```

**Comment by rianjs** [comment]
You also don't need to instantiate a `Calendar` inside a `using` statement:

``` csharp
using (Calendar calendar = new Calendar()) { }   //not necessary
```

I see this all over the place. Where does this usage come from?

**Comment by rianjs** [comment]
(I've closed the thread to minimize the number of open issues, but feel free to reply if you want to continue the conversation.)

**Comment by xrisdoc** [comment]
Thanks for looking at that! I replaced the line with what you had suggested, but this did not make any difference.

However, I have tested this further with another email account setup on the same version of Outlook and this seemed to work for this other account.

The email account where this is not working is my company email account which goes through our own Exchange server. So, I am going to check this out to see if something is happening to the emails when it is coming into our exchange server, that could be causing this.

**Comment by xrisdoc** [comment]
As for instantiating a `Calendar` within a using statement. `Calendar` seems to be implementing `IDisposable`, and I always try to create a new instance of an `IDisposable` object with the using statement, to ensure that it calls the `Dispose()` method. 

Are there any particular reasons why I shouldn't use the `using` statement for this?

**Comment by rianjs** [comment]
> Thanks for looking at that! I replaced the line with what you had suggested, but this did not make any difference.
> 
> However, I have tested this further with another email account setup on the same version of Outlook and this seemed to work for this other account.
> 
> The email account where this is not working is my company email account which goes through our own Exchange server. So, I am going to check this out to see if something is happening to the emails when it is coming into our exchange server, that could be causing this.

I suspect the problem is somewhere between the thing that generates the ics text, and the thing that delivers mail to the recipient. I know there are any number of ways to inspect and manipulate emails before they hit a mailbox, but if the ics text is valid, there's no much I can do.

I guess I would test by exporting the text using the code I provided, and then import it into the instance of Outlook that's giving you the error message. If Outlook is happy, you know the problem is somewhere in the mail delivery chain.

**Comment by rianjs** [comment]
> As for instantiating a Calendar within a using statement. Calendar seems to be implementing IDisposable, and I always try to create a new instance of an IDisposable object with the using statement, to ensure that it calls the Dispose() method.
> 
> Are there any particular reasons why I shouldn't use the using statement for this?

Wow, you're quite right, how did I miss that? There's no reason for it to implement `IDisposable`, so I deleted the method. All it did was clear the child objects, none of which was unmanaged. (Which means calling `Dispose` is superfluous.) It'll go out in the next version I publish to nuget.

(There are a lot of unmanaged `Stream`s floating around, but I think I've got all of them wrapped in `using` statements internally at the API boundary. I hope to get rid of some/all of them for v3. I have no idea why they're so prevalent.)
